### PR TITLE
feat(api/swagger): #11 - Adds basic swagger annotation with a sample …

### LINKED ISSUE #11: API: Swagger Annotation setup
Reference - https://github.com/ramsunvtech/trello-app-clone-spring-ng/pull/31

### PR DESCRIPTION
**Pull request by naveenvignesh5**
## Steps To Implement Feature

1. Add spring library for Open API 3.0 to `pom.xml`

```xml
<dependency>
    <groupId>org.springdoc</groupId>
    <artifactId>springdoc-openapi-ui</artifactId>
    <version>1.5.1</version>
</dependency>
```

2. Add custom path from Open API page to `application.properties`

```properties
springdoc.swagger-ui.path=/docs
```

3. Add a controller to simulate an API to test swagger api.

```java
package com.google.search.googlesearch.controller.v1.api;

import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RequestParam;
import org.springframework.web.bind.annotation.RestController;

import io.swagger.v3.oas.annotations.media.Content;
import io.swagger.v3.oas.annotations.media.Schema;
import io.swagger.v3.oas.annotations.responses.ApiResponse;
import io.swagger.v3.oas.annotations.responses.ApiResponses;

@RestController
public class SearchController {
    @GetMapping(value = "/api/v1/search")
    @ApiResponses(value = {
        @ApiResponse(responseCode = "200", description = "Get search results", content = @Content(schema = @Schema(nullable = true)))
    })
    public String getSearch(@RequestParam(defaultValue = "param") String query) {
        return "Query: " + query;
    }
}
```
resolves #11

### CODE REVIEW

File: api/google-search/pom.xml [hunk 1]
```diff
@@ -44,6 +45,11 @@
 			<artifactId>spring-boot-starter-test</artifactId>
 			<scope>test</scope>
 		</dependency>
+		<dependency>
+			<groupId>org.springdoc</groupId>
+			<artifactId>springdoc-openapi-ui</artifactId>
+			<version>1.5.1</version>
+		</dependency>
```
Comment (ramsunvtech) [inline api/google-search/pom.xml:52]: can we indent by 2 spaces through in vscode

File: api/google-search/src/main/resources/application.properties [hunk 2]
```diff
@@ -1 +1 @@
-
+springdoc.swagger-ui.path=/docs
```
Comment (ramsunvtech) [inline api/google-search/src/main/resources/application.properties:1]: keep all this config as well

```
springdoc.swagger-ui.path=/swagger
springdoc.swagger-ui.configUrl=/v3/api-docs
springdoc.swagger-ui.disable-swagger-default-url=false
```

**Review by ramsunvtech — CHANGES_REQUESTED** [review]

File: api/google-search/src/main/resources/application.properties (hunk 2 above)
Comment (naveenvignesh5) [reply api/google-search/src/main/resources/application.properties:3]: Added Properties.

File: api/google-search/pom.xml (hunk 1 above)
Comment (naveenvignesh5) [reply api/google-search/pom.xml:52]: Dependencies before the xml are also in same indentation. I hope this is the issue. If anything else kindly elaborate else kindly resolve.

**Review by shangan23 — APPROVED** [review]
Approved from my end.

**Review by ramsunvtech — APPROVED** [review]

**Review by jpukg — APPROVED** [review]
Looks good.

**Review by yuvarajuv — APPROVED** [review]
Approved from my end

**Review by karthikeyanpa12 — APPROVED** [review]

**Pull request merged**